utils/cmake/dreamcast.cmake

```diff
@@ -60,40 +79,10 @@ function(kos_add_romdisk target romdiskPath)
     add_custom_command(
         OUTPUT  ${obj}
         DEPENDS ${obj_tmp}
-        COMMAND ${CMAKE_C_COMPILER} -o ${obj} -r ${obj_tmp} -L$ENV{KOS_BASE}/lib/dreamcast -Wl,--whole-archive -lromdiskbase
+        COMMAND ${KOS_CC_BASE}/bin/sh-elf-gcc -o ${obj} -r ${obj_tmp} -L$ENV{KOS_BASE}/lib/dreamcast -Wl,--whole-archive -lromdiskbase
```

Comment:
I think you can use `${KOS_BASE}`, it should have been set by the main toolchain script.

Reply:
But that's just the KOS repo root path. We need the path to the toolchain folder, which is separate and usually one level up.

Reply:
I think we should use either $ENV{KOS...} or ${KOS...}, but not mixed them. From my understanding, as the cmake variables are defined in the toolchain.cmake and taken from env variable, I suggest we replace $ENV{KOS_BASE} by ${KOS_BASE} here, and modify the test line 16 to check the cmake variables instead of the env variables, wdyt?

Reply:
@maishuji Sorry for the late response. Had a lot going on... but yes, you're right! On it!

Reply:
Alrighty, should be done.